### PR TITLE
feat(module): add ability to generate a routing file for new modules

### PR DESCRIPTION
**Pull request by Brocco**

### CODE REVIEW

**Comment by hansl** [comment]
LGTM, but I'll block this until my e2e is in and you add a new e2e to test this. It's easy so shouldn't be a big deal ;)

**Comment by hansl** [comment]
Okay my e2e PR is in master now. Could you rebase and add an e2e?  It should be simple :)  Just add a new e2e for modules (I think there might be missing one already).

**Comment by Brocco** [comment]
@hansl I updated the PR with some e2e tests

File: tests/e2e/tests/generate/module/module-routing.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import {join} from 'path';
+import {ng} from '../../../utils/process';
+import {expectFileToExist} from '../../../utils/fs';
+
+
+export default function() {
+  const moduleDir = join(process.cwd(), 'src', 'app', 'test-module');
```
Comment (hansl) [inline tests/e2e/tests/generate/module/module-routing.ts]: avoid using `process.cwd()`. And you can simply use `src/app/test-module` as it doesn't matter if it's windows or not (for the expect functions).

**Comment by hansl** [comment]
LGTM

**Comment by jennyraj** [comment]
Does this enable route generation or it is still disabled?

**Comment by hansl** [comment]
It generates all the building block but not route generation. Inserting and making your routes is simpler now than it was before though.

**Comment by jennyraj** [comment]
Thanks you!

**Comment by chriscurnow** [comment]
@hansl, re

> It generates all the building block but not route generation. Inserting and making your routes is simpler now than it was before though.

It looks like this is included in the latest release but how do we use it?

**Comment by hansl** [comment]
`ng generate module MODULE_NAME --routing`.

**Pull request merged**